donkeycar/pipeline/types.py

```diff
@@ -70,22 +71,50 @@ class TubDataset(object):
     '''
 
     def __init__(self, config: Config, tub_paths: List[str],
-                 shuffle: bool = True) -> None:
+                 shuffle: bool = True, select: Optional[str] = None) -> None:
         self.config = config
         self.tub_paths = tub_paths
         self.shuffle = shuffle
+        self.select = None
         self.tubs: List[Tub] = [Tub(tub_path, read_only=True)
                                 for tub_path in self.tub_paths]
         self.records: List[TubRecord] = list()
+        if select:
+            # add '_index' to the allowed field list as this is
+            # auto-generated by the tub and not part of the inputs
+            fields = self.tubs[0].manifest.inputs + ['_index']
+            self.select = create_filter_string(select, fields)
 
     def train_test_split(self) -> Tuple[List[TubRecord], List[TubRecord]]:
-        print(f'Loading tubs from paths {self.tub_paths}')
+        msg = f'Loading tubs from paths {self.tub_paths}' + f' with filter ' \
+              f'{self.select}' if self.select else ''
+        print(msg)
         self.records.clear()
         for tub in self.tubs:
             for underlying in tub:
-                record = TubRecord(self.config, tub.base_path,
-                                   underlying=underlying)
-                self.records.append(record)
+                record = TubRecord(self.config, tub.base_path, underlying)
+                # eval string uses 'record' as variable so this parses
+                if not self.select or eval(self.select):
+                    self.records.append(record)
 
         return train_test_split(self.records, shuffle=self.shuffle,
                                 test_size=(1. - self.config.TRAIN_TEST_SPLIT))
+
+
+def create_filter_string(filter_text: str, inputs: List[str],
```

Comment:
This should be completely unnecessary.

Reply:
Not sure I understand what you mean... the logic is, that you type 'user/throttle > 0.1' and the function converts this into `record.underlying['user/throttle'] > 0.1` which is then applied to the object `record`.

Reply:
I am saying that the config file should contain something like:

```python
def filter_record(record: TubRecord):
   """
   Return True if you want the `record` to be included for training.
   """
    # defaults are to consider every record 
   return True

FILTER = filter_record
```

There is very little value in us parsing out strings to convert them to lambdas or invent a DSL.

Reply:
Got it, yeah we can do this.